spring-cloud-netflix-core/src/test/java/org/springframework/cloud/netflix/zuul/filters/route/apache/HttpClientRibbonCommandFactoryTest.java

```diff
@@ -94,13 +95,199 @@ public void testHystrixTimeoutValueRibbonTimeouts() throws Exception {
+		HttpClientRibbonCommandFactory ribbonCommandFactory = new HttpClientRibbonCommandFactory(springClientFactory, zuulProperties, new HashSet<ZuulFallbackProvider>());
+		RibbonCommandContext context = mock(RibbonCommandContext.class);
+		doReturn("service").when(context).getServiceId();
+		HttpClientRibbonCommand ribbonCommand = ribbonCommandFactory.create(context);
+		assertEquals(50, ribbonCommand.getProperties().executionTimeoutInMilliseconds().get().intValue());
+	}
+
+	@Test
+	public void testDefaultRibbonSetting() throws Exception {
+		SpringClientFactory springClientFactory = mock(SpringClientFactory.class);
+		ZuulProperties zuulProperties = new ZuulProperties();
+		RibbonLoadBalancingHttpClient loadBalancingHttpClient = mock(RibbonLoadBalancingHttpClient.class);
+		IClientConfig clientConfig = new DefaultClientConfigImpl();
+		doReturn(loadBalancingHttpClient).when(springClientFactory).getClient(anyString(),
+			eq(RibbonLoadBalancingHttpClient.class));
+		doReturn(clientConfig).when(springClientFactory).getClientConfig(anyString());
+		HttpClientRibbonCommandFactory commandFactory = new HttpClientRibbonCommandFactory(springClientFactory, zuulProperties, new HashSet<ZuulFallbackProvider>());
+		RibbonCommandContext context = mock(RibbonCommandContext.class);
+		doReturn("service").when(context).getServiceId();
+		HttpClientRibbonCommand ribbonCommand = commandFactory.create(context);
+		assertEquals(4000, ribbonCommand.getProperties().executionTimeoutInMilliseconds().get().intValue());
+	}
+
+	@Test
+	public void testRibbonTimeoutAndRibbonRetriesDefaultAndNameSpaceSetting() throws Exception {
+		ConfigurationManager.getConfigInstance().setProperty("service.test.ConnectTimeout", 1000);
+		ConfigurationManager.getConfigInstance().setProperty("service.test.ReadTimeout", 1000);
+		SpringClientFactory springClientFactory = mock(SpringClientFactory.class);
+		ZuulProperties zuulProperties = new ZuulProperties();
+		RibbonLoadBalancingHttpClient loadBalancingHttpClient = mock(RibbonLoadBalancingHttpClient.class);
+		IClientConfig clientConfig = new DefaultClientConfigImpl();
+		clientConfig.set(IClientConfigKey.Keys.ConnectTimeout, 100);
+		clientConfig.set(IClientConfigKey.Keys.ReadTimeout, 500);
+		doReturn(loadBalancingHttpClient).when(springClientFactory).getClient(anyString(),
+			eq(RibbonLoadBalancingHttpClient.class));
+		doReturn(clientConfig).when(springClientFactory).getClientConfig(anyString());
+		HttpClientRibbonCommandFactory ribbonCommandFactory = new HttpClientRibbonCommandFactory(springClientFactory, zuulProperties, new HashSet<ZuulFallbackProvider>());
+		RibbonCommandContext context = mock(RibbonCommandContext.class);
+		doReturn("service").when(context).getServiceId();
+		HttpClientRibbonCommand ribbonCommand = ribbonCommandFactory.create(context);
+		assertEquals(1200, ribbonCommand.getProperties().executionTimeoutInMilliseconds().get().intValue());
```

Comment:
Where is `1200` coming from?

Reply:
This case is to test the following situation:
```
service.test.ConnectTimeout = 1000
service.test.ReadTimeout = 1000
ribbon.ConnectTimeout = 100
ribbon.ReadTimeout = 500
```
Because `IClientConfig clientConfig = new DefaultClientConfigImpl();`  `NameSpace` is the default (ribbon)
So at this time the timeout is 1200